Consul/Connect.cs

```diff
@@ -53,6 +54,178 @@ public class CAConfig
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public string Type { get; set; } = "consul";
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public string Description { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public string LegacyID { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public Dictionary<string, string> LegacyMeta { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public DateTime LegacyCreateTime { get; set; } = DateTime.UtcNow;
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public DateTime LegacyUpdateTime { get; set; } = DateTime.UtcNow;
+    }
+
+    public class IntentionPermission
+    {
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public string Action { get; set; }
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public IntentionHTTPPermission HTTP { get; set; }
+    }
+
+    public class IntentionHTTPPermission
+    {
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public string PathExact { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public string PathPrefix { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public string PathRegex { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public List<string> Methods { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public List<IntentionHTTPHeaderPermission> Header { get; set; }
+
+    }
+
+    public class IntentionHTTPHeaderPermission
+    {
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public string Name { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public bool Present { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public string Exact { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public string Prefix { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public string Suffix { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public string Regex { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public bool Invert { get; set; }
+    }
+
+    /// <summary>
+    /// This handles the response for any operation carried out on the ServiceIntentionsEntry Model
+    /// </summary>
+    public class ServiceIntentionsEntryResponse
+    {
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public Guid ID { get; set; }
```

Comment:
We use `string` in other places, it is better to be consistent:
```suggestion
        public string ID { get; set; }
```